File: backend/pawnscan/src/main/java/edu/cit/garbo/pawnscan/service/AuthService.java
package edu.cit.garbo.pawnscan.service;

import edu.cit.garbo.pawnscan.dto.AuthResponse;
import edu.cit.garbo.pawnscan.dto.LoginRequest;
import edu.cit.garbo.pawnscan.dto.RegisterRequest;

public interface AuthService {

    AuthResponse register(RegisterRequest request);

    AuthResponse login(LoginRequest request);
}

File: mobile/app/src/main/java/com/cit/pawnscan/api/AuthService.kt
package com.cit.pawnscan.api

import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.POST

interface AuthService {
    @POST("/api/auth/register")
    fun register(@Body request: RegisterRequest): Call<AuthResponse>

    @POST("/api/auth/login")
    fun login(@Body request: LoginRequest): Call<AuthResponse>
}
